girder/molecules/molecules/models/geometry.py

```diff
@@ -0,0 +1,54 @@
+from bson.objectid import ObjectId
+
+from girder.models.model_base import AccessControlledModel
+from girder.constants import AccessType
+
+from .molecule import Molecule as MoleculeModel
+
+
+class Geometry(AccessControlledModel):
+
+    def __init__(self):
+        super(Geometry, self).__init__()
+
+    def initialize(self):
+        self.name = 'geometry'
+        self.ensureIndex('moleculeId')
+
+        self.exposeFields(level=AccessType.READ, fields=(
+            '_id', 'moleculeId', 'cjson', 'provenanceType', 'provenanceId'))
+
+    def validate(self, doc):
+        # If we have a moleculeId ensure it is valid.
+        if 'moleculeId' in doc:
+            mol = MoleculeModel().load(doc['moleculeId'], force=True)
+            doc['moleculeId'] = mol['_id']
+
+        return doc
+
+    def create(self, user, moleculeId, cjson, provenanceType=None,
+               provenanceId=None, public=False):
+        geometry = {
+            'moleculeId': moleculeId,
+            'cjson': cjson,
+            'creatorId': user['_id']
+        }
+
+        if provenanceType is not None:
+            geometry['provenanceType'] = provenanceType
+
+        if provenanceId is not None:
+            geometry['provenanceId'] = provenanceId
```

Comment:
What values can `provenanceType` take? Is it free formed or should it be in forced as an enumeration?

Reply:
It is currently just a free formed string, where I was thinking to have some specific keywords such as "calculation", where we would expect the "provenanceId" to be the calculation id.

But we can limit the options in an enum if that would be preferable.

Reply:
Ok, that  makes sense we can keep it free formed for now, we may want it as an enumeration in the future, so the data is a little more structured.

Reply:
Sounds good.

Another keyword I was thinking of using would be something like "openbabel conformer". That one might not have a calculation to link back to (unless we run it as a task), but maybe the ID can be indices for the different conformers.